core/templates/dev/head/components/top_navigation_bar/top_navigation_bar_directive.html

```diff
@@ -188,6 +188,11 @@
         </li>
       </ul>
     </li>
+
+    <li uib-dropdown ng-if="username == ''" class="oppia-navbar-clickable-dropdown">
```

Comment:
Ditto -- ng-if="!username" perhaps? Not sure.

If you're looking for '' specifically then perhaps use === instead of ==.